rfcs/2022-04-09-accessibility.md

```diff
@@ -0,0 +1,984 @@
+
+https://user-images.githubusercontent.com/35374649/170882365-47b9caa8-eff8-4572-a319-d0f63785f497.mov
+
+We can also use the [@react-aria/live-announcer](https://github.com/adobe/react-spectrum/tree/main/packages/%40react-aria/live-announcer) package which gives us programmatic way of triggering announcements with `announce()` function, by creating temporary live-regions.
+
+**Native:**
+
+In react-native we have a [announceForAccessibility](https://reactnative.dev/docs/accessibilityinfo#announceforaccessibility) API in AccessibilityInfo interface for announcing dynamic content:
+
+> [Demo](https://snack.expo.dev/K17t0cAtLj)
+
+```ts
+React.useEffect(() => {
+  AccessibilityInfo.announceForAccessibility('Some content to be announced');
+}, []);
+```
+
+https://user-images.githubusercontent.com/35374649/170883127-b81bc324-3d5c-4b9d-905f-14733fcc821d.mp4
+
+
+### ARIA Patterns
+
+> [Specification](https://www.w3.org/WAI/ARIA/apg/patterns/)
+
+**First rule of ARIA:**
+
+<details>
+  <summary>No ARIA is better than Bad ARIA</summary>
+  
+  <br />
+
+> For more info read - [No ARIA is better than Bad ARIA](https://www.w3.org/WAI/ARIA/apg/practices/read-me-first/)
+
+Incorrect ARIA misrepresents visual experiences, Misrepresented aria roles and properties can introduce confusion while using assistive technologies, because a `role` is a promise, when you say `role="button"` It's a promise that the author will also implement the JavaScript associated with it to provide the expected behaviour for a `button`.  
+ Using a `role` without fulfilling the promise of that role is similar to making a "Place Order" button that abandons an order and empties the shopping cart.
+
+</details>
+
+[ARIA Authoring Practices Guide (APG)](https://www.w3.org/WAI/ARIA/apg/), recommends approaches to help developers make widgets, navigation, and behaviors accessible using WAI-ARIA roles, states, and properties.
+
+In blade, the components that we are going to build should follow these patterns, And blade developers should implement proper aria roles, keyboard interaction following the documented patterns in APG
+
+All the common UI widgets are available with proper documentation about it's behaviour and interactions, for example:
+
+- [Accordion](https://www.w3.org/WAI/ARIA/apg/patterns/accordion/)
+- [Dialog](https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/)
+- [Checkbox](https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/)
+- [And more..](https://www.w3.org/WAI/ARIA/apg/patterns/)
+
+While building each blade component we should read upon these patterns and ensure they behave as expected.
+
+## Design Accessibility
+
+Designing with accessibility in mind is an important part of the process, ensuring that our designs have proper color contrast, visual hierearchy & legible text is important.
+
+#### Best practices when using color styles and making them accessible: <!-- omit in toc -->
+
+- Use colors which are dark enough and atleast pass the AA standards.
```

Comment:
@anuraghazra We can also attach this link here - [https://usecontrast.com/guide](https://usecontrast.com/guide)

Reply:
great resource! Thanks.